### DISCUSSION TITLE
Different return value for "custom" entry?

### DISCUSSION THREAD

**Discussion by ghost** [davatorium/rofi#2020] [Q&A]
For a script I am writing I need to know if a user has entered a "custom" value (i.e. one not in the list). Is is possible to do this?

I know there is `-no-custom`, but that completely disables custom entries, which isn't the behaviour I want. I need to detect custom entries that and entered using the normal `Enter` key.

I find info about the variable `ROFI_RETV`, but it doesn't seem to work. Running this for example:

`cat file.txt | rofi -i -p "Launcher" -dmenu`, and then `echo $ROFI_RETV`, it's blank

The return value `$?` is 0 regardless of if it's a custom entry or not

Is there a way to detect custom entries via return value, or in some other way? I could use `-format i` and then `-1` means a custom entry, but that complicates my script

Edit:

The current best idea I have is to use `-format "i - s"`, and then read the first two chracters, if they are `-1`, it's a custom entry. Then delete the index and hyphen.

**Comment by cortsf** [comment]
> For a script I am writing I need to know if a user has entered a "custom" value (i.e. one not in the list). Is is possible to do this?


If I understand your problem, I think you can use the same language you are using to call rofi (bash?) to verify if the value returned by rofi is an element of the set of values you used to call rofi. You don't need rofi to tell you that.

Example (no particular programming language..)
```
vals = [ "aa", "bb", "cc" ]
value_returned_by_rofi = call rofi using $vals

if $value_returned_by_rofi is_elem_of $vals then
  print "yay!"
else
  print "ney!"
endif
```

**Comment by cortsf** [comment]
> The return value $? is 0 regardless of if it's a custom entry or not

This seems to be a bug. I'm not a rofi dev/expert but I'm using this feature heavily. Example using `kb-custom-1` to accept the first entry


```
[cortsf@lenovo:~]$ echo "hello,bye" | rofi -sep "," -dmenu
hello

[cortsf@lenovo:~]$ echo $?
10

```


Edit: After reading again your comment and thinking more about it, I think you are not understanding `rofi -dmenu` return values.

From the [docs](https://github.com/davatorium/rofi/blob/next/doc/rofi-dmenu.5.markdown):


> RETURN VALUE
> 0: Row has been selected accepted by user.
> 1: User cancelled the selection.
> 10-28: Row accepted by custom keybinding.


So yes, it always returns 0 if you accept your entry with `kb-accept-entry`
